File: c/JosephusProblem.c
#include <bits/stdc++.h>
using namespace std;

int jos(int a, int b){
	if(a == 1)
		return a;

	return (jos(a-1,b)+b-1)%a+1;
}


int main() {  
	int a = 7,b=2;
	cout<<jos(a,b);
	return 0; 
} 

File: java/JosephusProblem.java
import java.io.*;

class Algorithms{

static int jos(int n, int k){
	if (n == 1)
	   return 1;

	return (jos(n - 1, k) + k-1) % n + 1;


public static void main(String[] args){
	System.out.println(jos(6, 2));
    }
}

